modules/collect/src/main/java/com/opengamma/strata/collect/io/CharSources.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (C) 2017 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.collect.io;
+
+import java.io.File;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.nio.file.Path;
+
+import com.google.common.base.Charsets;
+import com.google.common.io.CharSource;
+import com.google.common.io.Files;
+import com.google.common.io.MoreFiles;
+import com.google.common.io.Resources;
+
+/**
+ * Helper that allows {@code CharSource} objects to be created.
+ */
+public final class CharSources {
+
+  private CharSources() { }
+
+  /**
+   * Obtains an instance of {@link CharSource} from a file name, specified as a String.
+   *
+   * @param  fileName the file name, as a String
+   * @return  a new instance of {@link CharSource} with UTF-8 for charset.
+   */
+  public static CharSource ofFileName(String fileName) {
+    return Files.asCharSource(new File(fileName), Charsets.UTF_8);
+  }
+
+  /**
+   * Obtains an instance of {@link CharSource} from a file name, specified as a String.
+   * This also takes in a specific character set, as a {@link Charset}.
+   *
+   * @param  fileName the file name, as a String
+   * @param  charset the charset to build the new CharSource based on
+   * @return  a new instance of {@link CharSource}
+   */
+  public static CharSource ofFileName(String fileName, Charset charset) {
+    return Files.asCharSource(new File(fileName), charset);
+  }
+
+  //---------------------------------------------------------------------------------------------
+  /**
+   * Obtains an instance of {@link CharSource} from a file object, specified as a {@link File}.
+   *
+   * @param  file the file object
```

Comment:
`@param  file the file object` -> `@param file  the file object` Throughout file